test/integ/test_cursor.py

```diff
@@ -609,15 +609,119 @@ def test_geography(conn, db_parameters):
             cnx.cursor().execute(f"drop table {name_geo}")
 
 
+@pytest.mark.skipolddriver
 def test_callproc(conn_cnx):
-    """Callproc test.
+    """Callproc test."""
 
-    Notes:
-        It's a nop as of now.
-    """
+    def create_stored_procedure(cursor):
+        cursor.execute(
+            "drop procedure if exists test_stored_procedure(varchar, int, date)"
+        )
+        cursor.execute("drop procedure if exists test_stored_procedure(float, char)")
+        cursor.execute("drop procedure if exists test_stored_procedure()")
+
+        cursor.execute(
+            """
+            create or replace procedure test_stored_procedure(p1 varchar, p2 int, p3 date)
+            returns string not null
+            language sql
+            as
+            begin
+              return 'teststring';
+            end;
+            """
+        )
+
+        cursor.execute(
+            """
+            create or replace procedure test_stored_procedure(p1 float, p2 char)
+            returns float not null
+            language sql
+            as
+            begin
+              return 1.23;
+            end;
+            """
+        )
+
+        cursor.execute(
+            """
+            create or replace procedure test_stored_procedure()
+            returns boolean
+            language sql
+            as
+            begin
+              return true;
+            end;
+            """
+        )
+
+    def verify_stored_procedure(cursor):
+        ret = cursor.callproc("test_stored_procedure", ("str", 1, "2022-02-22"))
+        assert ret == ("str", 1, "2022-02-22")
+        res = cur.fetchall()
+        assert len(res) == 1 and len(res[0]) == 1 and res[0][0] == "teststring"
+
+        ret = cursor.callproc("test_stored_procedure", (0.99, "c"))
+        assert ret == (0.99, "c")
+        res = cur.fetchall()
+        assert len(res) == 1 and len(res[0]) == 1 and res[0][0] == 1.23
+
+        ret = cursor.callproc("test_stored_procedure")
+        assert ret == ()
+        res = cur.fetchall()
+        assert len(res) == 1 and len(res[0]) == 1 and res[0][0] is True
+
+    # test pyformat paramstyle
     with conn_cnx() as cnx:
-        with pytest.raises(errors.NotSupportedError):
-            cnx.cursor().callproc("whatever the stored procedure")
+        with cnx.cursor() as cur:
+            create_stored_procedure(cur)
+            verify_stored_procedure(cur)
+
+    # test qmark paramstyle
+    with conn_cnx(paramstyle="qmark") as cnx:
+        with cnx.cursor() as cur:
+            create_stored_procedure(cur)
+            verify_stored_procedure(cur)
```

Comment:
Instead of defining helper functions, or copy pasting tests I like to [parametrize tests](https://docs.pytest.org/en/6.2.x/parametrize.html) instead

Reply:
indeed, thanks for calling it out, updated